spring-kafka/src/main/java/org/springframework/kafka/support/CorrelatedProducerListener.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2015-2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.kafka.support;
+
+import org.apache.kafka.clients.producer.RecordMetadata;
+
+/**
+ * Listener for handling outbound Kafka messages. Exactly one of its methods will be invoked, depending on whether
+ * the write has been acknowledged or not.
+ *
+ * Returns any arbitrary correlation data provided by the caller.
+ *
+ * Its main goal is to provide a stateless singleton delegate for {@link org.apache.kafka.clients.producer.Callback}s,
+ * which, in all but the most trivial cases, requires creating a separate instance per message.
+ *
+ * @param <K> the key type.
+ * @param <V> the value type.
+ *
+ * @author Marius Bogoevici
+ * @author Gary Russell
+ * @since 1.3
+ *
+ * @see org.apache.kafka.clients.producer.Callback
+ */
+public interface CorrelatedProducerListener<K, V> {
```

Comment:
Why do we need this at all since `send(ProducerRecord<K, V> record, ProducerListener<K, V> producerListener)` does exactly what we need for correlation request with the async response?

More over there is already mentioned `org.apache.kafka.clients.producer.Callback`...